core/procs.go

```diff
@@ -1103,23 +1103,43 @@ var procReadString Proc = func(args []Object) Object {
 	return readFromReader(strings.NewReader(EnsureString(args, 0).S))
 }
 
-func readLine(r *BufferedReader) (s string, err error) {
-	for {
-		line, isPrefix, err := r.ReadLine();
-		if err != nil {
-			return "", err
+func readLine(inp io.Reader) (s string, e error) {
+	switch r := inp.(type) {
+	case *BufferedReader: // Underlying type is *bufio.Reader
+		for {
+			line, isPrefix, err := r.ReadLine();
+			if err != nil {
+				return "", err
+			}
+			s += string(line)
+			if !isPrefix {
+				break
+			}
 		}
-		s += string(line)
-		if !isPrefix {
-			break
+	case *Buffer: // Underlying type is *bytes.Buffer
+		s, e = r.ReadString('\n')
```

Comment:
Would not this code work for `bufio.Reader` as well (since it also has `ReadString` method)? If so, `readLine` could be simplified by accepting an interface with `ReadString` method...